app/test/specs/mainControllerSpec.js

```diff
@@ -79,34 +80,36 @@
             expect(state.go).toHaveBeenCalledWith('app.institution', {institutionKey: '1239'});
         });
 
-        it('Should call state.go() and InstitutionService.getInstitution() in function goToSearchedInstitution()', function(){
-            spyOn(state, 'go').and.callThrough();
-            spyOn(instService, 'getInstitution').and.callThrough();
-            httpBackend.expect('GET', "/api/institutions/" + splab.key).respond(splab);
-            mainCtrl.goToSearchedInstitution(splab.key);
+        it('Should call makeSearch() in function showMenu()', function(){
+            var documents = [{name: splab.name, id: splab.key}];
+            mainCtrl.search = splab.name;
+            mainCtrl.finalSearch = mainCtrl.search;
+            spyOn(mainCtrl, 'makeSearch').and.callThrough();
+            spyOn(instService, 'searchInstitutions').and.callThrough();
+            spyOn(mainCtrl, 'openMenu');
+            httpBackend.expect('GET', "api/search/institution?name=" + splab.name + "&state=active").respond(documents);
+            mainCtrl.showMenu('$event').then(function success() {
```

Comment:
testando promise em um teste síncrono é incorreto, sempre que for testar um método assincrono faça um teste assíncrono, usando function(done) e no final done() ou .should.notify(done)